File: src/Other/TwoSum2.java
package Other;

import java.util.Arrays;

// https://leetcode.com/problems/two-sum-ii-input-array-is-sorted/
public class TwoSum2 {
    public int[] twoSum(int[] numbers, int target) {
        int i = 0;
        int j = numbers.length - 1;

        while (i < j) {
            if (numbers[i] + numbers[j] > target) {
                j--;
            } else if (numbers[i] + numbers[j] < target) {
                i++;
            } else {
                return new int[]{i + 1, j + 1};
            }
        }
        return new int[2];
    }

    public static void main(String[] args) {
        TwoSum2 t = new TwoSum2();
        int[] numbers = {2, 7, 11, 15};
        System.out.println(Arrays.toString(t.twoSum(numbers, 9)));
    }

}

File: py/essential/TwoSum2.py
# https://leetcode.com/problems/two-sum-ii-input-array-is-sorted/description/?envType=problem-list-v2&envId=rr2ss0g5
# Neet

from typing import List


class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        # Two pointers
        left, right = 0, len(numbers) - 1
        while left < right:
            current_sum = numbers[left] + numbers[right]
            if current_sum == target:
                return [left + 1, right + 1]
            elif current_sum < target:
                left += 1
            else:
                right -= 1
        return []


s = Solution()
print(s.twoSum([2, 7, 11, 15], 9))  # [1,2]
print(s.twoSum([2, 3, 4], 6))  # [1,2]
